src/date-field.tsx

```diff
@@ -10,8 +10,9 @@ import {isFirefox} from './utils/platform';
 
 import type {CommonFormFieldProps} from './text-field-base';
 
-export interface DateFieldProps extends CommonFormFieldProps {
```

Comment:
maybe we could make `CommonFormFieldProps` generic and have something like `CommonFormFieldProps<HTMLInputElement>`, we could even make `HTMLInputElement` the default generic value, so `CommonFormFieldProps` would be equivalent to `CommonFormFieldProps<HTMLInputElement>`, in case we need a textarea we'd use `CommonFormFieldProps<HTMLTextAreaElement>`